commit 7373ed3c9ff77c5f1de864eabfcd44b8742b9488
Author: Jaime Frey <jfrey@cs.wisc.edu>
Date:   Thu Oct 1 16:35:01 2015 -0500

    Fix NETWORK_HOSTNAME config parameter. #5288
    
    In init_local_hostname_impl(), when NETWORK_HOSTNAME is set, the string
    that we pass to getipaddr() was left empty. When getipaddr() fails, we
    would then retry it repeatedly for 1 minute.

diff --git a/src/condor_utils/ipv6_hostname.cpp b/src/condor_utils/ipv6_hostname.cpp
index 080994e..7a8c644 100644
--- a/src/condor_utils/ipv6_hostname.cpp
+++ b/src/condor_utils/ipv6_hostname.cpp
@@ -47,7 +47,6 @@ bool init_local_hostname_impl()
 		dprintf(D_HOSTNAME, "NETWORK_HOSTNAME says we are %s\n", local_hostname.Value());
 	}
 
-	MyString test_hostname;
 	if( ! local_hostname_initialized ) {
 		// [TODO:IPV6] condor_gethostname is not IPv6 safe. Reimplement it.
 		char hostname[MAXHOSTNAMELEN];
@@ -57,10 +56,11 @@ bool init_local_hostname_impl()
 					"local hostname, ip address, FQDN.\n");
 			return false;
 		}
-		test_hostname = hostname;
-		local_hostname = test_hostname;
+		local_hostname = hostname;
 	}
 
+	MyString test_hostname = local_hostname;
+
 	bool local_ipaddr_initialized = false;
 	bool local_ipv4addr_initialized = false;
 	bool local_ipv6addr_initialized = false;
